### PR TITLE
add PQconnectdb to Query Logger

### PR DESCRIPTION
**Pull request by MorriganR**
Befor:

After:

### CODE REVIEW

File: src/providers/postgres/qgspostgresconn.cpp [hunk 1]
```diff
@@ -394,7 +397,9 @@ QgsPostgresConn::QgsPostgresConn( const QString &conninfo, bool readOnly, bool s
         break;
       }
 
+      QString errorMsg = PQerrorMessage();
```
Comment (elpaso) [inline src/providers/postgres/qgspostgresconn.cpp:400]:
```suggestion
      const QString errorMsg = PQerrorMessage();
```

File: src/providers/postgres/qgspostgresconn.cpp [hunk 2]
```diff
@@ -418,11 +424,13 @@ QgsPostgresConn::QgsPostgresConn( const QString &conninfo, bool readOnly, bool s
   {
     QString errorMsg = PQerrorMessage();
     PQfinish();
+    logWrapper->setError( errorMsg );
     QgsMessageLog::logMessage( tr( "Connection to database failed" ) + '\n' + errorMsg, tr( "PostGIS" ) );
     mRef = 0;
     return;
   }
 
+  logWrapper = nullptr;
```
Comment (elpaso) [inline src/providers/postgres/qgspostgresconn.cpp:433]:
```suggestion
```

this is not required

**Review by elpaso — APPROVED** [review]

File: src/providers/postgres/qgspostgresconn.cpp (hunk 2 above)
Comment (MorriganR) [reply src/providers/postgres/qgspostgresconn.cpp:433]: this is necessary for the correct calculation of the execution time...

Comment (elpaso) [reply src/providers/postgres/qgspostgresconn.cpp:433]: Ok, I didn't notice the rest of the function.

File: src/providers/postgres/qgspostgresconn.cpp [hunk 3]
```diff
@@ -405,6 +410,7 @@ QgsPostgresConn::QgsPostgresConn( const QString &conninfo, bool readOnly, bool s
       QgsDebugMsgLevel( "Connecting to " + mUri.connectionInfo( false ), 2 );
       QString connectString = mUri.connectionInfo();
       addDefaultTimeoutAndClientEncoding( connectString );
+      logWrapper = std::make_unique<QgsDatabaseQueryLogWrapper>( QStringLiteral( "PQconnectdb()" ), connectString.toUtf8(), QStringLiteral( "postgres" ), QStringLiteral( "QgsPostgresConn" ), QGS_QUERY_LOG_ORIGIN_PG_CON );
```
Comment (elpaso) [inline src/providers/postgres/qgspostgresconn.cpp:413]: Maybe useful to add some suffix to "PQconnectdb()"  to differentiate it from the other log entry at the beginning of the function?

Comment (MorriganR) [reply src/providers/postgres/qgspostgresconn.cpp:413]: "libpq::PQconnectdb()" - ?

**Review by nyalldawson — APPROVED** [review]

**Pull request merged**